### ISSUE TITLE
Add a `getFrontmostApp` method to the API

### ISSUE THREAD

**Issue by Dru89** [raycast/extensions#1797]
**Describe the feature and the current behavior/state.**
This new method would provide an [`Application`](https://developers.raycast.com/api-reference/utilities#application) reference to whatever the foreground application is.

Today, if you want to access this information you need to do something like:
```ts
import { runAppleScript } from "run-applescript"
import { getApplications } from "@raycast/api";

const appsPromise = getApplications();
const frontmostPromise = runAppleScript(
  'tell application "System Events" to return the bundle identifier of the first application process where frontmost is true'
);

const [apps, frontmost] = await Promise.all([appsPromise, frontmostPromise]);
const frontmostApp = apps.find((app) => app.bundleId === frontmost);
```

This is both error prone, and slow since we need to start a new subprocess inside of the running JavaScript. (See "Any Other Info")

**Will this change the current API? How?**
I'd propose adding another method to your existing [utilities](https://developers.raycast.com/api-reference/utilities). You already have an interface for [`Application`](https://developers.raycast.com/api-reference/utilities#application) that we can reuse here, so probably a method with a signature like:

```ts
async function getFrontmostApplication(): Promise<Application>;
```

**Who will benefit from this feature?**
There are a at least a couple of reasons you might want to get the frontmost app:
- Controlling/reading the state of the app with AppleScript/JXA. (Example: Getting a URL from the active browser, quitting the active application, moving the active window of the active application to the left half of the screen.)
- Capturing the name of the app as context for any data the user might be recording. (Example: The clipboard history in Raycast shows the source of the app the content was captured from. Not *quite* the same API, but the same idea at least.)

**Any Other info.**
The Raycast API already offers APIs that are spiritually similar, allowing the user to get *content* from the frontmost application:
- [`getSelectedText()`](https://developers.raycast.com/api-reference/environment#getselectedtext) returns the highlighted text of the foreground application.
- [`getSelectedFinderItems()`](https://developers.raycast.com/api-reference/environment#getselectedfinderitems) returns any selected items in a finder window.

All of these attributes (including getting the frontmost application) would be accessible through AppleScript/JXA, but getting them natively from an API means that we wouldn't need to:
1. Wait for the command to launch.
2. Then start a subprocess.
3. Have that subprocess run `osascript` with some provided (potentially buggy) AppleScript
4. Wait for the result to complete, then return it to the shell.
5. Have the shell pipe that back out of the subprocess
6. And then resolve that in the upper command.

Running the AppleScript *itself* isn't prohibitively slow, but if many people use it then it could be prone to more bugs across multiple implementations. And the true win here would be the lack of subprocess creation, which *could* take a few hundred milliseconds (maybe up to a second) to complete. That kind of latency could be prohibitive if you need that information to bootstrap the command.

---

There is *one* gotcha that I can think of to selecting this data (and potentially all the better reason to make it an internal API).

Consider the following AppleScript:
```applescript
tell application "System Events" to return the name of the first application process where frontmost is true
```

What this *actually* does is look for the name of the running application process. This may *not* be the same as the running application **name**. For instance, Visual Studio Code returns "Electron", because the application process is `/Applications/Visual Studio Code.app/Contents/MacOS/Electron`.

I've found returning the `bundle identifier` to be a more reliable indicator of the running application. (In the example above, `com.microsoft.VSCode` is returned.) I know that `bundleId` is listed as an *optional* property on `Application`, though, so it may be more beneficial to use bundle identifier as the primary lookup, and app name as the fallback. (Or maybe you have something even better that you can do natively on your end. 😄)

**Comment by georgesnow** [comment]
'''''AppleScript 

tell application "System Events" to return the **short** name of the first application process where frontmost is true

''''

I agree api would be nice but if modify the above AppleScript to "short" name. It will return the app name instead of bundle id

And technically you can't shorten the script and use without the tell block. I just tried with this last week actually.



'''''AppleScript

return the **short** name of the first application process where frontmost is true

''''

**Comment by georgesnow** [comment]
````applescript 

tell application "System Events" to return the  short name of the first application process where frontmost is true

````

I agree api would be nice but if modify the above AppleScript to "short" name. It will return the app name instead of bundle id

And technically you can't shorten the script and use without the tell block. I just tried with this last week actually.



````applescript

return short name of (info for (path to frontmost application))

`````

**Comment by denisidoro** [comment]
Thanks for the suggestion, @georgesnow.

That said, it takes ~1.5s from shelling out / calling osascript to getting this return value.

A native function would hopefully be much faster.

**Comment by georgesnow** [comment]
So I found this to be a little bit more performant while we wait for native method. I am not typescript expert use with caution, but it works. I am have been trying it with an extension  for raycast.

```javascript
const { 
exec } = require('child_process');
const 
bundleidcmd = `echo 'import AppKit;var bundleid = NSWorkspace.shared.frontmostApplication?.bundleIdentifier ?? "Unknown";print(bundleid)' | swift -`;
const 
bundleid = exec (`${bundleidcmd}`)
bundleid.stdout.on('data', (stuff:any)
=> {
console.log(`stdout:
${stuff}`);
// const idder=${data};
});
```

**Comment by mathieudutour** [comment]
It's now available in v1.44.0!
